client/src/components/Layout/Events/Events.js

```diff
@@ -21,23 +24,50 @@ class Events extends Component {
       .catch((err) => console.log(err))
   }
 
+  changeInput =  (name) => {
+    this.setState({ eventName: name.target.value.substr(0,20)
+       })
+  }
+
   render () {
-    const { events } = this.state
+  const { events, eventName} = this.state
+  const filtertsearch = events.filter(
+    (event) =>
+    {
+   return event.event_title.toLowerCase().indexOf(eventName.toLowerCase())!== -1
+    }
+  )
     return (
+
       <div>
         <NavBar />
+        <div className='SearchEvent'>
+          <div>
+            <input type='text'
+              placeholder='Search for Event ... '
+              onChange={this.changeInput.bind(this)}
```

Comment:
the binding shouldn't be done here, but in the constructor. In fact, since `changeInput` is defined through a class field syntax (`changeInput = (name) => ...`), the binding is not necessary and can be dropped altogether. See https://reactjs.org/docs/handling-events.html for more details.
In any case, here it should be just `onChange={this.changeInput}`